geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestSecurityIntegrationTest.java

```diff
@@ -62,61 +58,58 @@ public static void before() throws Exception {
   }
 
   @Test
-  public void testFunctions() throws Exception {
-    String json = "{\"@type\":\"double\",\"@value\":210}";
-
-    HttpResponse response = restClient.doGet("/functions", "unknown-user", "1234567");
+  public void testListFunctions() throws Exception {
+    HttpResponse response = restClient.doGet("/functions", "user", "wrongPswd");
     assertEquals(401, GeodeRestClient.getCode(response));
-    response = restClient.doGet("/functions", "stranger", "1234567");
+    response = restClient.doGet("/functions", "user", "user");
     assertEquals(403, GeodeRestClient.getCode(response));
-    response = restClient.doGet("/functions", "dataReader", "1234567");
+    response = restClient.doGet("/functions", "dataRead", "dataRead");
     assertEquals(200, GeodeRestClient.getCode(response));
     response.getEntity();
     assertThat(GeodeRestClient.getContentType(response))
         .containsIgnoringCase(MediaType.APPLICATION_JSON_UTF8_VALUE);
+  }
 
-    response = restClient.doPost("/functions/AddFreeItemsToOrder", "unknown-user", "1234567", json);
+  @Test
+  public void executeFunction() throws Exception {
+    HttpResponse response =
+        restClient.doPost("/functions/invalid-function-id", "user", "wrongPswd", "");
     assertEquals(401, GeodeRestClient.getCode(response));
-    response = restClient.doPost("/functions/AddFreeItemsToOrder", "dataReader", "1234567", json);
-    assertEquals(403, GeodeRestClient.getCode(response));
-    response = restClient.doPost("/functions/AddFreeItemsToOrder?onRegion=" + REGION_NAME,
-        "dataWriter", "1234567", json);
-    // because we're only testing the security of the endpoint, not the endpoint functionality, a
-    // 500 is acceptable
-    assertEquals(500, GeodeRestClient.getCode(response));
+    response = restClient.doPost("/functions/invalid-function-id", "user", "user", "");
+    assertEquals(404, GeodeRestClient.getCode(response));
```

Comment:
Shouldn't this be a "403"? Test failed when I tried running it.

Reply:
you will need to rebuild the war file before running the test.

Reply:
Thanks, my oversight.